app/src/main/java/com/realityexpander/tasky/agenda_feature/data/common/workers/RefreshAgendaWeekWorker.kt

```diff
@@ -77,27 +79,49 @@ class RefreshAgendaWeekWorker @AssistedInject constructor(
     }
 
     companion object {
-        const val NOTIFICATION_ID = 100001
-        const val WORKER_NAME = "REFRESH_AGENDA_WEEK_WORKER"
+        private const val WORKER_NAME = "REFRESH_AGENDA_WEEK_WORKER"
+        private const val NOTIFICATION_ID = 100001
 
         const val PARAMETER_START_DATE = "startDate"
         const val START_DAY_OFFSET = -5
         const val END_DAY_OFFSET = 5
-    }
 
-    init {
-        WorkerNotifications.createNotificationChannel(
-            context,
-            WORKER_NOTIFICATION_CHANNEL_ID,
-            context.getString(R.string.agenda_sync_refresh_worker_human_readable_notification_channel)
-        )
+        // • Start the one-time Refresh 'Agenda Week' Worker
+        fun startWorker(applicationContext: Context) {
+            val refreshAgendaWeekConstraints: Constraints = Constraints.Builder().apply {
+                setRequiredNetworkType(NetworkType.CONNECTED)
+            }.build()
+            val data = Data.Builder()
+                .putString(
+                    RefreshAgendaWeekWorker.PARAMETER_START_DATE,
+                    ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS).toString()
+                )
+                .build()
+            val name = RefreshAgendaWeekWorker.WORKER_NAME
+            val agendaWeekWorkRequest =
+                OneTimeWorkRequestBuilder<RefreshAgendaWeekWorker>()
+                    .setConstraints(refreshAgendaWeekConstraints)
+                    .setInputData(data)
+                    .addTag(name)
+                    .addTag("For 10 days around ${data.getString(RefreshAgendaWeekWorker.PARAMETER_START_DATE)}")
+                    .addTag(TASKY_WORKERS_TAG)
+                    .setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
+                    .setBackoffCriteria(BackoffPolicy.LINEAR, 1, TimeUnit.MINUTES)
+                    .build()
+            WorkManager.getInstance(applicationContext)
```

Comment:
Why are you creating a worker inside a worker?

Can't this worker just call a method to do it?
Or what's the point of having this worker create a new worker?